File: frontend/src/core/models/recipe.js
export class Recipe {
  constructor({ id, title = "", description = "", ingredients = [], servings = 0, minutes = 0 } = {}) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.ingredients = ingredients.slice();
    this.servings = servings;
    this.minutes = minutes;
  }
}

File: backend/internal/model/recipe.go
package model

// Recipe mirrors the `Recipe` entity (RecipeDTO). Servings/minutes are *int so
// they marshal as null when unset (spec: nullable).
type Recipe struct {
	ID          int      `json:"id"`
	Title       string   `json:"title"`
	Description string   `json:"description"`
	Ingredients []string `json:"ingredients"`
	Servings    *int     `json:"servings"`
	Minutes     *int     `json:"minutes"`
}